aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/endpoints/OperationInputTestGenerator.kt

```diff
@@ -0,0 +1,172 @@
+import software.amazon.smithy.rust.codegen.client.smithy.generators.clientInstantiator
+import software.amazon.smithy.rust.codegen.core.rustlang.Attribute
+import software.amazon.smithy.rust.codegen.core.rustlang.AttributeKind
+import software.amazon.smithy.rust.codegen.core.rustlang.CargoDependency
+import software.amazon.smithy.rust.codegen.core.rustlang.escape
+import software.amazon.smithy.rust.codegen.core.rustlang.join
+import software.amazon.smithy.rust.codegen.core.rustlang.rust
+import software.amazon.smithy.rust.codegen.core.rustlang.rustBlock
+import software.amazon.smithy.rust.codegen.core.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.core.rustlang.writable
+import software.amazon.smithy.rust.codegen.core.smithy.RustCrate
+import software.amazon.smithy.rust.codegen.core.smithy.generators.setterName
+import software.amazon.smithy.rust.codegen.core.testutil.integrationTest
+import software.amazon.smithy.rust.codegen.core.testutil.tokioTest
+import software.amazon.smithy.rust.codegen.core.util.dq
+import software.amazon.smithy.rust.codegen.core.util.expectMember
+import software.amazon.smithy.rust.codegen.core.util.inputShape
+import software.amazon.smithy.rust.codegen.core.util.nonEmptyOrNull
+import software.amazon.smithy.rust.codegen.core.util.orNull
+import software.amazon.smithy.rust.codegen.core.util.toSnakeCase
+import java.util.logging.Logger
+
+class OperationInputTestDecorator : ClientCodegenDecorator {
+    override val name: String = "OperationInputTest"
+    override val order: Byte = 0
+
+    override fun extras(codegenContext: ClientCodegenContext, rustCrate: RustCrate) {
+        val endpointTests = EndpointTypesGenerator.fromContext(codegenContext).tests.nonEmptyOrNull() ?: return
+        rustCrate.integrationTest("endpoint_tests") {
+            Attribute(Attribute.cfg("feature = \"${TestUtilFeature.name}\"")).render(this, AttributeKind.Inner)
+            val tests = endpointTests.flatMap { test ->
+                val generator = OperationInputTestGenerator(codegenContext, test)
+                test.operationInputs.filterNot { usesUnsupportedBuiltIns(it) }.map { operationInput ->
+                    generator.generateInput(operationInput)
+                }
+            }
+            tests.join("\n")(this)
+        }
+    }
+}
+
+private val deprecatedBuiltins =
+    setOf(Builtins.S3_USE_GLOBAL_ENDPOINT, Builtins.STS_USE_GLOBAL_ENDPOINT).map { it.builtIn.get() }
+
+fun usesUnsupportedBuiltIns(testOperationInput: EndpointTestOperationInput): Boolean {
+    return testOperationInput.builtInParams.members.map { it.key.value }.any { deprecatedBuiltins.contains(it) }
+}
+
+/**
+ * Eventually, we need to pull this test into generic smithy. However, this relies on generic smithy clients
+ * supporting middleware and being instantiable from config (https://github.com/awslabs/smithy-rs/issues/2194)
+ *
+ * Doing this in AWS codegen allows us to actually integration test generated clients.
+ */
```

Comment:
I removed the second `*` since I wouldn't classify this as an actual doc comment.
```suggestion
/*
 * TODO(https://github.com/awslabs/smithy-rs/issues/2194) Eventually, we need to pull this test into
 * generic smithy. However, this relies on generic smithy clients supporting middleware and being
 * instantiable from config 
 *
 * Doing this in AWS codegen allows us to actually integration test generated clients.
 */
```